pkg/ruler/rulestore/bucketclient/bucket_client.go

```diff
@@ -102,9 +102,16 @@ func (b *BucketRuleStore) ListAllRuleGroups(ctx context.Context) (map[string]rul
 
 	// List rule groups for all tenants.
 	err := b.bucket.Iter(ctx, "", func(key string) error {
+		// Ignore objects that look like directories, ie. they have empty namespace or group name.
+		// Such objects can exist in the object store, but would cause trouble when loading them, because
+		// we check for user, namespace and group components not being empty.
+		if strings.HasSuffix(key, objstore.DirDelim) {
+			return nil
+		}
+
 		userID, namespace, group, err := parseRuleGroupObjectKeyWithUser(key)
```

Comment:
What if we actually change `parseRuleGroupObjectKeyWithUser()` and return an error if the group is empty?

Reply:
We should really check for user, namespace or group being non-empty.

Reply:
I thought it wasn't possible to have empty user and namespace due to how the parsing work but, if it's possible, then definitely yet.